0.12/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM ubuntu:18.04
+
+ENV THRIFT_VERSION v0.12.0
+
+RUN buildDeps=" \
+		automake \
+		bison \
+		curl \
+		flex \
+		g++ \
+		libboost-dev \
+		libboost-filesystem-dev \
+		libboost-program-options-dev \
+		libboost-system-dev \
+		libboost-test-dev \
+		libevent-dev \
+		libssl-dev \
+		libtool \
+		make \
+		pkg-config \
+	"; \
+	apt-get update && apt-get install -y --no-install-recommends $buildDeps && rm -rf /var/lib/apt/lists/* \
+	&& curl -k -sSL "https://github.com/apache/thrift/archive/${THRIFT_VERSION}.tar.gz" -o thrift.tar.gz \
+	&& mkdir -p /usr/src/thrift \
+	&& tar zxf thrift.tar.gz -C /usr/src/thrift --strip-components=1 \
+	&& rm thrift.tar.gz \
+	&& cd /usr/src/thrift \
+	&& ./bootstrap.sh \
+	&& ./configure  --without-python --without-cpp \
+	&& make \
+	&& make install \
+	&& cd / \
+	&& rm -rf /usr/src/thrift \
+	&& curl -k -sSL "https://storage.googleapis.com/golang/go1.4.linux-amd64.tar.gz" -o go.tar.gz \
+	&& tar xzf go.tar.gz \
+	&& rm go.tar.gz \
+	&& cp go/bin/gofmt /usr/bin/gofmt \
+	&& rm -rf go \
+	&& apt-get purge -y --auto-remove $buildDeps
+
+CMD [ "thrift" ]
```

Comment:
It would be better if this container did not run thrift as root.

There's a good recipe for running inside a container as a regular user in the [Boost Docker Development Environment](https://github.com/jeking3/bdde).